metric/system/process/process_common.go

```diff
@@ -33,9 +34,42 @@ import (
 	sysinfo "github.com/elastic/go-sysinfo"
 )
 
-// ProcsMap is a map where the keys are the names of processes and the value is the Process with that name
+//ProcsMap is a convinence wrapper for the oft-used ideom of map[int]ProcState
 type ProcsMap map[int]ProcState
 
+// ProcsTrack is a thread-safe wrapper for a process Stat object's internal map of processes.
+type ProcsTrack struct {
+	pids ProcsMap
+	mut  sync.Mutex
+}
+
+func NewProcsMap() *ProcsTrack {
+	return &ProcsTrack{
+		pids: make(ProcsMap, 0),
+		mut:  sync.Mutex{},
```

Comment:
```suggestion
```
I don't think this is necessary, `mut sync.Mutex` should be usable immediately since it is equivalently default constructed.